src/components/item/RelatedUnits.vue

```diff
@@ -0,0 +1,437 @@
+      'isItemFormInSidebar',
+    ]),
+    chosenItemIndex() {
+      return this.itemsFromPossibleRelatedList
+        .findIndex(item => item.id === this.chosenItemId);
+    },
+    chosenListIndex() {
+      return this.lists
+        .findIndex(list => list.id === this.chosenListId);
+    },
+    isAddRelatedButtonDisabled() {
+      return this.newRelatedUnitType === 'item' 
+        ? !this.chosenItemId
+        : !this.chosenListId;
+    },
+    choosableLists() {
+      return this.lists
+        .filter(list => this.isUnitChoosable(list, 'listId', 'relatedLists'));
+    },
+    choosableItems() {
+      return this.itemsFromPossibleRelatedList
+        .filter(item => this.isUnitChoosable(item, 'id', 'relatedItems'));
+    },
+    alreadyRelatedItemsInChosenList() {
+      return this.currentItemObj?.relatedItems
+        ?.filter(item => item.listId === this.chosenListId);
+    },
+  },
+  watch: {
+    chosenListId(value) {
+      this.itemsFromPossibleRelatedList = [];
+
+      const listIndex = this.lists.findIndex(list => list.id === value);
+
+      if (this.newRelatedUnitType === 'item' && this.lists[listIndex]?.items?.[0] instanceof Object) {
+        this.itemsFromPossibleRelatedList = this.lists[listIndex].items;
+      } else if (value && this.newRelatedUnitType === 'item') {
+        this._fetchItemsByListId({
+          id: value,
+          cancelToken: null,
+        })
+          .then(responseList => {
+            this.itemsFromPossibleRelatedList = responseList.items;
+          });
+      }
+    },
+  },
+  methods: {
+    ...mapMutations([
+      'updateItemFieldLocally',
+      'updateItemRelatedUnitsLocally',
+      'resetRelatedUnitsLocally',
+    ]),
+    ...mapActions([
+      '_fetchItemsByListId',
+      '_updateItemOnServer',
+      '_addItemOnServer',
+    ]),
+    toggleShowingStatus(target) {
+      this.showingStatuses[target] = !this.showingStatuses[target];
+    },
+    changeNewRelatedUnitType(value) {
+      this.newRelatedUnitType = value;
+      this.chosenListId = null;
+      this.chosenItemId = null;
+    },
+    setNewRelatedUnitId(unitType, value) {
+      unitType === 'list' 
+        ? this.chosenListId = value 
+        : this.chosenItemId = value;
+    },
+    isUnitChoosable(unit, unitIdFieldName, targetRelatedArr) {
+      const isItemAddingMode = this.newRelatedUnitType === 'item';
+      const isCurrentUnitList = !unit.listId;
+      const isListChoosingOnlyToShowItems = isItemAddingMode && isCurrentUnitList;
+
+      const isSavedOnServer = typeof unit.id !== 'undefined';
+      const isItself = this.edittingItemObj[unitIdFieldName] === unit.id;
+      const isAlreadyRelated = this.edittingItemObj[targetRelatedArr]?.includes(unit.id);
+
+      const isChoosable = isSavedOnServer && !isItself && !isAlreadyRelated;
+
+      return isListChoosingOnlyToShowItems ? isSavedOnServer : isChoosable;
+    },
+    updateItemField({ field, stringsArray, objectsArray }) {  
```

Comment:
stringArray => idsForApiUpdate
objectArray => fullUnitsForStoreUpdate

for example